Italian Translation Update

MishaBoar opened the pull request:
Full italian translation of all missing phrases. Some notes:

* I shifted to using "portafoglio" instead of "portamonete" to refer to "wallet" as this is what is traditionally used in other wallets and somehow feels more correct or current to me.
* For the term "dust" I opted for the more literal Italian translation "polvere" as it is how this is commonly referred to on italian tutorials/websites. I added (dust) next to polvere to allow people to be able to understand that "dust" RPC commands are referring to the same concept.
* Where possible, I referred to specific terms (as in portafoglio vs portamonete) used also on Bitcoin Core, unless I noticed the terms used were not very sound in Italian.

I have been working on this with a colleague (who in the past provided translation for the italian dogecoin website) for a while, so we tried to double check many times. I will keep this updated and do further tests as 1.14.6 release approaches.

patricklodder commented:
Thanks!

I'm adding "help wanted" in the hopes that someone else that speaks Italian can give us a second pair of eyes on this. It'd be appreciated! ❤️

MishaBoar commented:
Absolutely, it is much needed to have (at least!) another pair of eyes and ears to eventually further improve and adjust this. Thanks!

MishaBoar commented:
For reviewers: I am committing later today a revision with minor changes and some adjustments (in particular to parts I had not translated myself) as I totally forced my (also Italian) partner to use a build of Dogecoin Core with the new translation for the past 2 weeks. I used @chromatic's new tool, as well, which allowed to fix/update a few more phrases.

MishaBoar commented:
So, to bring the discussion here on Github, as I asked this to @chromatic - we have two files for italian, https://github.com/dogecoin/dogecoin/blob/1.14.6-dev/src/qt/locale/bitcoin_it.ts (the main language file) and https://github.com/dogecoin/dogecoin/blob/1.14.6-dev/src/qt/locale/bitcoin_it_IT.ts.  It generates some confusion as well in the language selector, as I am sure most Italians would be puzzled by the fact of having two Italian languages available for selection.

As there are no countries with their own specific flavor of the written Italian language, what is the purpose? Or is there a technical reason for this which forces us to keep it? Due to the peculiar history of the written Italian language (it is basically a codified and enforced standard defined originally by the literary tradition of Tuscany, which anyhow differs significantly from local actually spoken dialects), there is really only one "Standard Italian". Of course one could have Dogecoin lovingly translated into local regional dialects, but some of those does not even descend directly from Italian.

The file was added with this commit (https://github.com/dogecoin/dogecoin/blob/1.14.6-dev/src/qt/locale/bitcoin_it_IT.ts): https://github.com/dogecoin/dogecoin/commit/4cbe05bf205ceeb23d1d47360693c8235b497ed5

patricklodder commented:
> what is the purpose?

None because there is no `it_CH`. I recommend to remove `it_IT` (separate commit please) and just maintain `it`.

MishaBoar commented:
OK, done!

patricklodder commented:
@MishaBoar you'll need to delete the following line from the qt makefile (`src/Makefile.qt.include`) for it to no longer expect the it_IT translation too: 

https://github.com/dogecoin/dogecoin/blob/10c294d2eb6cfeab1537eabc616762ca7debf1b4/src/Makefile.qt.include#L52

Sorry, I forgot to mention that above!

MishaBoar commented:
Building fine, working great.

![Capture](https://user-images.githubusercontent.com/92135099/166856903-f41395cd-303b-4ab6-a0cb-676e9cf2c7d1.jpg)
t

DogecoinItalia commented:
> Full italian translation of all missing phrases. Some notes:
> 
> * I shifted to using "portafoglio" instead of "portamonete" to refer to "wallet" as this is what is traditionally used in other wallets and somehow feels more correct or current to me.
> * For the term "dust" I opted for the more literal Italian translation "polvere" as it is how this is commonly referred to on italian tutorials/websites. I added (dust) next to polvere to allow people to be able to understand that "dust" RPC commands are referring to the same concept.
> * Where possible, I referred to specific terms (as in portafoglio vs portamonete) used also on Bitcoin Core, unless I noticed the terms used were not very sound in Italian.
> 
> I have been working on this with a colleague (who in the past provided translation for the italian dogecoin website) for a while, so we tried to double check many times. I will keep this updated and do further tests as 1.14.6 release approaches.

Credo che il termine “spiccioli” sia il termine che più si avvicina a “dust”

MishaBoar commented:
> > Full italian translation of all missing phrases. Some notes:
> > 
> > * I shifted to using "portafoglio" instead of "portamonete" to refer to "wallet" as this is what is traditionally used in other wallets and somehow feels more correct or current to me.
> > * For the term "dust" I opted for the more literal Italian translation "polvere" as it is how this is commonly referred to on italian tutorials/websites. I added (dust) next to polvere to allow people to be able to understand that "dust" RPC commands are referring to the same concept.
> > * Where possible, I referred to specific terms (as in portafoglio vs portamonete) used also on Bitcoin Core, unless I noticed the terms used were not very sound in Italian.
> > 
> > I have been working on this with a colleague (who in the past provided translation for the italian dogecoin website) for a while, so we tried to double check many times. I will keep this updated and do further tests as 1.14.6 release approaches.
> 
> Credo che il termine “spiccioli” sia il termine che più si avvicina a “dust”

Grazie!

Yes, it would sound better. The problem in this case "dust" is a technical term referring to a very specific thing (a transaction lower than a certain minimum amount) https://www.investopedia.com/terms/b/bitcoin-dust.asp), whereas "spiccioli" would just seem to refer to a more "generic" small amount left into the wallet. 

Additionally, there is a problem when we have "technical" terms like this one as there is already an existing literature online about this, in Italian, that refers to it as "polvere" (usually in relation to Bitcoin, from which Dogecoin is derived). I also added "dust" between parentheses in this specific case to allow people to refer to the technical term in English as this seems a very specific case where this might be helpful to the end user due, once again, to existing literature.

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -154,107 +154,107 @@
     <message>
         <source>Warning: If you encrypt your wallet and lose your passphrase, you will &lt;b&gt;LOSE ALL OF YOUR DOGECOINS&lt;/b&gt;!</source>
-        <translation>Attenzione: se si cifra il portamonete e si perde la passphrase &lt;b&gt;TUTTI I PROPRI Dogecoin ANDRANNO PERSI&lt;/b&gt;!</translation>
+        <translation>Attenzione: se si cripta il portafoglio e si perde la password &lt;b&gt;TUTTI I DOGECOIN ANDRANNO PERSI&lt;/b&gt;!</translation>
```
Attenzione: se cripti il portafoglio e perdi la password, PERDERAI TUTTI I TUOI DOGECOIN

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -154,107 +154,107 @@
     <message>
         <source>The passphrase entered for the wallet decryption was incorrect.</source>
-        <translation>La frase inserita per decrittografare il tuo portafoglio è incorretta</translation>
+        <translation>La password inserita per decrittografare il tuo portafoglio è incorretta</translation>
```
La password inserita per decrittare il tuo portafoglio è errata

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -280,15 +280,15 @@
     <message>
         <source>Show general overview of wallet</source>
-        <translation>Mostra lo stato generale del portamonete</translation>
+        <translation>Mostra lo stato generale del portafoglio</translation>
```
Panoramica del portafoglio

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -352,15 +352,15 @@
     <message>
         <source>Backup wallet to another location</source>
-        <translation>Effettua il backup del portamonete</translation>
+        <translation>Effettua il backup del portafoglio</translation>
```
Esegui il backup del portafoglio in un'altra posizione

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -400,15 +400,15 @@
     <message>
         <source>Encrypt the private keys that belong to your wallet</source>
-        <translation>Cifra le chiavi private che appartengono al tuo portamonete</translation>
+        <translation>Cripta le chiavi private che appartengono al tuo portafoglio</translation>
```
Cripta le chiavi private del tuo portafoglio

MishaBoar replied:
These are all the phrases we inherited from core when I swapped from portamonete to portafoglio, They do need some massaging... Thanks!

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -548,78 +548,78 @@
     <message>
         <source>Print paper wallets</source>
-        <translation type="unfinished"></translation>
+        <translation>Stampa portafogli di carta</translation>
```
Stampa paper wallets

MishaBoar replied:
I thought about this for a long, long time. While I agree with you that for younger people or people that know English this sounds much more correct, for the average user that does the extra effort to switch to Italian, "paper wallet" might be a bit too advanced. I imagine my uncle or relatives that would have no clue about what either word mean. "Portafogli di carta" is used also in other wallets for Ethereum/Bitcoin/Litecoin.

To be brief, younger people with modern education in English might not even make the effort to switch to the Italian language; those who do, are those that could have more problems understanding English. Let me know ;)

DogecoinItalia replied:
quando un utente si trova di fronte la parola sconosciuta "paper wallet" o non attinente "portafoglio cartaceo" la prima cosa che farà sarà quella di cercare il significato su google. cercando paper wallet troverà subito la risposta più pertinente.

MishaBoar replied:
Sei piu' ottimista di me ;) Proviamo!

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -910,63 +910,63 @@
     <message>
         <source>Invalid private key; please check and try again!</source>
-        <translation type="unfinished"></translation>
+        <translation>Chiave privata non valida; controlla e prova di nuovo!</translation>
```
Chiave privata non valida; controlla e riprova!

MishaBoar replied:
More compact version, I like

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -910,63 +910,63 @@
     <message>
         <source>Invalid address generated from private key; please check and try again!</source>
-        <translation type="unfinished"></translation>
+        <translation>L&apos;indirizzo generato dalla chiave privata non è valido; controlla e prova di nuovo!</translation>
```
L&apos;indirizzo generato dalla chiave privata non è valido; controlla e riprova!

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -985,7 +985,7 @@
     <message>
         <source>%1 will download and store a copy of the Dogecoin block chain. At least %2GB of data will be stored in this directory, and it will grow over time. The wallet will also be stored in this directory.</source>
-        <translation>%1 scaricherà e salverà una copia della Blockchain di Dogecoin. Saranno salvati almeno %2GB di dati in questo percorso e continueranno ad aumentare col tempo. Anche il portafoglio verrà salvato in questo percorso.</translation>
+        <translation>%1 scaricherà e salverà una copia della block chain di Dogecoin. Saranno salvati almeno %2GB di dati in questo percorso e continueranno ad aumentare col tempo. Anche il portafoglio verrà salvato in questo percorso.</translation>
```
%1 scaricherà e salverà una copia della blockchain di Dogecoin. Saranno salvati almeno %2GB di dati in questa directory e continueranno ad aumentare col tempo. Anche il portafoglio verrà salvato in questa directory.

MishaBoar replied:
Yep, good for consistency. Maybe the original in English should be switched as well.

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -1277,7 +1277,7 @@ Per specificare più URL separarli con una barra verticale &quot;|&quot;.</trans
+        <translation>&amp;Lingua interfaccia utente:</translation>
     </message>
     <message>
         <source>The user interface language can be set here. This setting will take effect after restarting %1.</source>
```
Qui puoi scegliere la lingua dell'interfaccia utente. La scelta avrà effetto dopo il riavvio.

MishaBoar replied:
like it

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -1408,118 +1408,118 @@ Per specificare più URL separarli con una barra verticale &quot;|&quot;.</trans
     <message>
         <source>Helpful tip of the day:</source>
-        <translation type="unfinished"></translation>
+        <translation>Suggerimento utile del giorno:</translation>
```
Suggerimento del giorno:

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -1531,103 +1531,103 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>The total exceeds your balance when the transaction fee is included</source>
-        <translation type="unfinished"></translation>
+        <translation>Il totale inclusivo delle commissioni di transazione è superiore al saldo disponibile</translation>
```
Il totale, inclusa la commissione di transazione, è superiore al saldo disponibile.

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -1531,103 +1531,103 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>The amount exceeds your balance.</source>
-        <translation type="unfinished">Non hai abbastanza fondi</translation>
+        <translation>Non hai abbastanza fondi</translation>
```
Saldo insufficiente

MishaBoar replied:
Nicer, should be suggested also to Bitcoin Core.

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -1531,103 +1531,103 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>added as transaction fee</source>
-        <translation type="unfinished"> Includi il costo della transazione</translation>
+        <translation> aggiunti come commissioni di transazione</translation>
```
Aggiunti per la commissione di transazione

MishaBoar replied:
La frase "2 dogecoin aggiunti come comissioni di transazione" mi suona piu' corretto, ma non lo difenderei a spada tratta. Usiamo la tua versione

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -2202,55 +2202,55 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>Ban for</source>
-        <translation type="unfinished"></translation>
+        <translation>Bandisci per</translation>
```
Banna per

MishaBoar replied:
I did not dare to, thanks

DogecoinItalia replied:
altrimenti blocca /sblocca

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -2202,55 +2202,55 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>WARNING: Scammers have been active, telling users to type commands here, stealing their wallet contents. Do not use this console without fully understanding the ramification of a command.</source>
-        <translation type="unfinished"></translation>
+        <translation>ATTENZIONE: Truffatori potrebbero tentare di convincere utenti inesperti ad utilizzare comandi che permettono di rubare il contenuto del loro portafoglio. Non usare questa console senza essere a completa conoscenza dei rischi insiti in un comando.</translation>
```
Truffatori potrebbero tentare di convincere utenti inesperti ad utilizzare comandi che permettono di rubare il contenuto del loro portafoglio. Non usare questa console se non conosci alla perfezione il rischio che comporta l'utilizzo di ciascun comando.

MishaBoar replied:
More scorrevole, I like it

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -2202,55 +2202,55 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>Are you sure you want to remove the peer: </source>
-        <translation type="unfinished"></translation>
+        <translation>Sei sicuro di volere rimuovere il peer: </translation>
```
Sei sicuro di volere rimuovere il peer? :

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -2574,7 +2574,7 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>Copy dust</source>
-        <translation>Copia trascurabile</translation>
+        <translation>Copia polvere</translation>
```
copia cifra irrisoria

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -2622,71 +2622,71 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>If the custom fee is set to 1000 satoshis and the transaction is only 250 bytes, then &quot;per kilobyte&quot; only pays 250 satoshis in fee, while &quot;total at least&quot; pays 1000 satoshis. For transactions bigger than a kilobyte both pay by kilobyte.</source>
-        <translation type="unfinished"></translation>
+        <translation>Se le commissioni di transazione sono configurate per essere 1000 satoshi e la transazione é solo 250 bytes, allora, per kilobyte, la commissione sará solamente di 250 satoshi, mente il totale sará di almeno 1000 satoshi. Per transazioni più grandi di un kilobyte entrambi pagano per kilobyte.</translation>
```
Se la commissione personalizzata è impostata su 1000 satoshi e la transazione é solo 250 bytes, la commissione sará solamente di 250 satoshi per kilobyte, mentre il totale sará di almeno 1000 satoshi. Per transazioni più grandi di un kilobyte entrambi pagano per kilobyte.

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -2622,71 +2622,71 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>The total exceeds your balance when the %1 transaction fee is included.</source>
-        <translation type="unfinished"></translation>
+        <translation>Il totale è superiore al saldo disponibile quando le commissioni di transazione di %1 sono incluse.</translation>
```
Il totale, inclusa la commissione di transazione di %1, è superiore al saldo disponibile.

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -2622,71 +2622,71 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>Duplicate address found: addresses should only be used once each.</source>
-        <translation type="unfinished"></translation>
+        <translation>Indirizzo già inserito: lo stesso indirizzo può essere inserito solo una volta.</translation>
```
<translation>Indirizzo già inserito: non puoi riutilizzare lo stesso indirizzo.</translation>

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -2622,71 +2622,71 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>The address you selected for change is not part of this wallet. Any or all funds in your wallet may be sent to this address. Are you sure?</source>
-        <translation type="unfinished"></translation>
+        <translation>L&apos;indirizzo che hai selezionato come indirizzo di scambio non é parte di questo portafoglio. Parte o tutti i tuoi fondi potrebbero essere inviati a questo indirizzo. Sei sicuro?</translation>
```
<translation>L&apos;indirizzo che hai selezionato come indirizzo di scambio non appartiene a questo portafoglio. I  tuoi fondi potrebbero essere in parte o totalmente inviati a questo indirizzo. Sei sicuro?</translation>

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -2769,7 +2769,7 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>Enter a label for this address to add it to your address book</source>
-        <translation type="unfinished"></translation>
+        <translation>Inserisci un&apos;etichetta per questo indirizzo da aggiungere alla tua rubrica</translation>
```
<translation>Inserisci un&apos;etichetta per questo indirizzo per aggiungerlo alla tua rubrica</translation>

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -3118,52 +3118,52 @@ p, li { white-space: pre-wrap; }
         </translation>
     </message>
     <message numerus="yes">
         <source>matures in %n more block(s)</source>
```
cosa si intende per matures?

MishaBoar replied:
E' un termine tecnico, letteralmente tradotto con "maturare" in Italiano. Indica il momento (dopo un certo di numero di blocchi) in cui una transazione puo' essere confermata. This might help: https://bitcoin.stackexchange.com/questions/23957/transaction-maturity-vs-block-maturity

I think the translation is correct, as it relies on the same metaphor - the one around maturity - used for the english counterpart.

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -3590,7 +3590,7 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>Delete all wallet transactions and only recover those parts of the blockchain through -rescan on startup</source>
-        <translation>Elimina tutte le transazioni dal portamonete e recupera solo quelle che fanno parte della blockchain attraverso il comando -rescan all&apos;avvio.</translation>
+        <translation>Elimina tutte le transazioni dal portafoglio e recupera solo quelle che fanno parte della blockchain attraverso il comando -rescan all&apos;avvio.</translation>
```
<translation>Elimina tutte le transazioni dal portafoglio e recupera solo quelle che fanno parte della blockchain usando il comando -rescan all&apos;avvio.</translation>

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -4310,187 +4310,187 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>Sets the serialization of raw transaction or block hex returned in non-verbose mode, non-segwit(0) or segwit(1) (default: %d)</source>
-        <translation type="unfinished"></translation>
+        <translation>Imposta la serializzazione delle transazioni raw e dei blocchi esadecimali in modalitá non-verbosa, non-segwit (0) o segwit (1) (default: %d)</translation>
```
<translation>Imposta l'ordine delle transazioni raw e dei blocchi esadecimali in modalitá non-verbosa, non-segwit (0) o segwit (1) (default: %d)</translation>

MishaBoar replied:
La traduzione originale e' corretta. Serializzazione e' un termine tecnico che indica il comporre un oggetto complesso in maniera lineare  in modo che possa essere communicato o salvato in memoria. "Ordine" non sarebbe corretto in questo caso.

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -4310,187 +4310,187 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>Warning: The network does not appear to fully agree! Some miners appear to be experiencing issues.</source>
-        <translation type="unfinished"></translation>
+        <translation>Attenzione: La rete sembra essere in disaccordo! Alcuni miner sembrano avere problemi.</translation>
```
<translation>Attenzione: La rete sembra essere in disaccordo! Pare che alcuni miner stiano riscontrando dei problemi.</translation>

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -4310,187 +4310,187 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>Warning: We do not appear to fully agree with our peers! You may need to upgrade, or other nodes may need to upgrade.</source>
-        <translation type="unfinished"></translation>
+        <translation>Attenzione: Impossibile accordarsi con i peers! Potrebbe essere necessario aggiornare il nodo, o gli altri nodi potrebbero dover aggiornare il proprio software.</translation>
```
<translation>Attenzione: Impossibile accordarsi con i peers! Potrebbe essere necessario aggiornare il software del tuo o degli altri nodi.</translation>

DogecoinItalia commented on src/qt/locale/bitcoin_it.ts:
```diff
@@ -4310,187 +4310,187 @@ p, li { white-space: pre-wrap; }
     <message>
         <source>The wallet will avoid paying less than the minimum relay fee.</source>
-        <translation type="unfinished"></translation>
+        <translation>Il portafoglio preverrá il pagamento di commissioni inferiori alle commissioni minime</translation>
```
<translation>Il portafoglio eviterà il pagamento di commissioni inferiori alle commissioni minime</translation>

MishaBoar replied:
Hmmm maybe. I am bit on the fence on this one but we can go for your version.

michilumin commented:
Ok. Adding my review on this based ONLY on two years of university level Italian I took and years of living in New Jersey near the Italian American districts around guys named Lil Tony (i'm not making this up), but this is pretty basic language and seriously it seems right to me from my exposure.

MishaBoar commented:
Folks, thank you. Integrating @DogecoinItalia's suggestions (check also my comments). The commit will be up soon.

MishaBoar commented:
Integrated suggestions in the last commit as per @DogecoinItalia's comments above; test build seems fine!

patricklodder commented:
Thanks all for the thorough reviewing! I've removed the "help wanted" label.

michilumin reviewed (approved):
Yay more languages.  Several looks by @DogecoinItalia @MishaBoar and myself. Che bello!
